code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AutoCarouselNotificationBuilder.kt

```diff
@@ -63,19 +63,16 @@ internal object AutoCarouselNotificationBuilder {
 
         // fallback to a basic push template notification builder if less than 3 images were able to be downloaded
         if (downloadedImageCount < PushTemplateConstants.DefaultValues.CAROUSEL_MINIMUM_IMAGE_COUNT) {
-            Log.trace(
-                LOG_TAG, SELF_TAG,
-                "Less than 3 images are available for the auto carousel push template, falling back to a basic push template."
-            )
+            Log.trace(LOG_TAG, SELF_TAG, "Less than 3 images are available for the auto carousel push template, falling back to a basic push template.")
+            val intent = AEPPushNotificationBuilder.createIntent("FallbackAction", pushTemplate)
```

Comment:
This seems like an overhead to create an intent with all the keys just because we don't want to store `messageData`. Is there any reason why we don't want to have it in AEPPushTemplate?

Reply:
I remembered now why we didn't want to save `messageData` as map. But I still don't see the need to create an intent here. Can BasicNotificationBuilder.fallbackToBasicNotification accept `NotificationData`? That way we can store `messageData` as `NotificationData` as well

Reply:
Above will require adding `putString` to `NotificationData` and its implementation in `IntentData` and `MapData`

Reply:
I am inclined to keeping
NotificationData - to be just a reader object and not able to modify the data
Template instance - to be immutable as well 

This is a good problem to solve.